web/source/keyboards/kmwkeyboards.ts

```diff
@@ -1081,22 +1082,25 @@ namespace com.keyman.keyboards {
         try {
           if(result instanceof Error) {
             promiseFuncs.reject(result as Error);
-          } else { // TODO (#5044): pass `result` to the `resolve` function.
-            promiseFuncs.resolve();
+          } else if (promiseFuncs) {
```

Comment:
This condition should be unnecessary.  I'd double-check other aspects of your implementation if it appears to be needed, as if so... you should also need it for the `if` section as well.